src/EnergyPlus/Boilers.hh

```diff
@@ -197,15 +182,12 @@ namespace Boilers {
 
 struct BoilersData : BaseGlobalStruct
 {
-    int numBoilers = 0;
     bool getBoilerInputFlag = true;
     Array1D<Boilers::BoilerSpecs> Boiler;
 
     void clear_state() override
     {
-        this->numBoilers = 0;
-        this->getBoilerInputFlag = true;
-        this->Boiler.deallocate();
```

Comment:
Why do we no longer have to call `deallocate()`? Did we not have to call it to begin with?

Reply:
So ... what is happening to the old instance? Does it get properly destroyed or are we leaking insane amounts of memory here? I admit I don't have a lot of experience with the `*this =` idiom. It looks very strange and wrong to me.

Reply:
This should not be leaking anything.  I made a prototype of this and called `clear_state` 50bn times and the memory usage on the process never went up, and I could see the destructor being called every time, so the assignment seems to be doing everything we need here.  This paradigm was suggested by @lefticus, so I feel pretty good about it.

Now, that said, I think once we have all the `clear_state` functions cleaned up to this level, we can just go back to where we are calling `clear_state`, and change the call so that we don't need the intermediary `clear_state` function, if in fact all it is doing is deleting the last one and creating a new one.

(As a final aside, just note that this function is only called during serial unit test runs, and in case someone is calling clear_state during serial runs of E+ via API)